debugger/src/main.rs

```diff
@@ -85,20 +90,69 @@ enum CliCommands {
 }
 
 fn main() -> Result<()> {
-    tracing_subscriber::fmt::fmt()
-        .compact()
-        .with_ansi(false)
-        .without_time()
-        .with_env_filter(EnvFilter::from_default_env())
-        .with_writer(stderr)
-        .init();
+    let log_info_message = setup_logging()?;
 
     let matches = CliCommands::parse();
 
     match matches {
         CliCommands::List {} => list_connected_devices()?,
         CliCommands::ListChips {} => list_supported_chips()?,
-        CliCommands::Debug { port, vscode } => debug(port, vscode)?,
+        CliCommands::Debug { port, vscode } => debug(port, vscode, &log_info_message)?,
     }
     Ok(())
 }
+
+/// Setup logging, according to the following rules.
+/// 1. If the RUST_LOG environment variable is set, use it as a `LevelFilter` to configure a subscriber that logs to a file in the system's application data directory.
+/// 2. Irrespective of the RUST_LOG environment variable, configure a subscribe that will write with `LevelFilter::ERROR` to stderr, because these errors are picked up and reported to the user by the VSCode extension.
+fn setup_logging() -> Result<String, anyhow::Error> {
```

Comment:
Could we put this inte to `probe-rs-cli-util` and share with other utilities? :)

Reply:
I agree in principle, but request we move this to a new discussion to identify and prioritize overlaps(multiple implementations of same capability) in various probe-rs organization tools.

Reply:
Sounds sensible!